src/container/srv_epoch.c

```diff
@@ -136,6 +136,13 @@ ds_cont_epoch_aggregate(struct rdb_tx *tx, struct ds_pool_hdl *pool_hdl,
 		DP_CONT(pool_hdl->sph_pool->sp_uuid, in->cei_op.ci_uuid), rpc,
 		in->cei_epoch);
 
+	/* Verify handle has write access */
+	if (!ds_sec_cont_can_write_data(hdl->ch_sec_capas)) {
+		D_ERROR(DF_CONT": permission denied to aggregate\n",
+			DP_CONT(cont->c_svc->cs_pool_uuid, cont->c_uuid));
+		return -DER_NO_PERM;
```

Comment:
Not urgent, but maybe better for this function would be to use the rc variable and ensure in all error and success cases that the D_DEBUG at the end of the function is run, to provide more information when logging is enabled.

Reply:
Hm, yeah, this is inconsistent with the rest of the file. I think I was looking at the check immediately below, where we do a direct return. Will clean this up if there's anything else to add.

Reply:
Fixing